src/shogun/mathematics/linalg/backend/eigen/BasicOps.cpp

```diff
@@ -208,6 +226,46 @@ void LinalgBackendEigen::add_diag_impl(
 	A_eig.diagonal() = alpha * A_eig.diagonal() + beta * b_eig;
 }
 
+template <typename T>
+void LinalgBackendEigen::pinvh_impl(
+    const SGMatrix<T>& A, SGMatrix<T>& result) const
+{
+	auto m = A.num_rows;
+	SGVector<T> s(m);
+	SGMatrix<T> V(m, m);
+	typename SGMatrix<T>::EigenMatrixXtMap A_eig = A;
+	typename SGMatrix<T>::EigenMatrixXtMap result_eig = result;
+	
+	eigen_solver_symmetric_impl<T>(A, s, V, m);
+	
+	typename SGVector<T>::EigenVectorXtMap s_eig = s;
+	typename SGMatrix<T>::EigenMatrixXtMap V_eig = V;
+		
+	float64_t pinv_tol = CMath::MACHINE_EPSILON * m * s_eig.real().maxCoeff();
+	s_eig.array() = (s_eig.real().array()>pinv_tol).select(s_eig.real().array().inverse(), 0);
```

Comment:
maybe this line can be shared somewhere between both pinv versions? it is the same code after all